### DISCUSSION TITLE
[@types/react-csv] CommonPropTypes.data can be a function

### DISCUSSION THREAD

**Discussion by fpintos** [DefinitelyTyped/DefinitelyTyped#60853] [Issues with a @types package] [answered]
According to https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react-csv/components/CommonPropTypes.d.ts, the `data` field passed to CSVLink or CSVDownload is typed as `string | object[]`, but according to https://github.com/react-csv/react-csv we can pass a function that returns the data.

Looking at the [CSVLink implementation](https://github.com/react-csv/react-csv/blob/master/src/components/Link.jsx) and their [prop definition](https://github.com/react-csv/react-csv/blob/master/src/metaProps.js), I see that data can be a function when onClick and asyncOnClick are passed as well. One _could_ split CommonPropTypes and make it so the compiler will detect when you are calling with all the correct parameters and types.

On the other hand, it does not look like CSVDownload supports `data` being a function, both components share CommonPropTypes, so it would be necessary to make data as a function only for CSVLink.

[a comment by an automated account is not shown]

**Comment by martin-badin** [comment, marked as answer]
PR is ready https://github.com/DefinitelyTyped/DefinitelyTyped/pull/60860
